recipe/activate.bat

```diff
@@ -1,3 +1,20 @@
+@@echo off
+setlocal enabledelayedexpansion
+
+:: save existing variables for deactivation script; initialize all backups to a placeholder
+:: (empty is the same as undefined), and only overwrite them if a value exists
+for %%X in (
+    CC CXX CMAKE_ARGS CMAKE_GEN CMAKE_GENERATOR CMAKE_GENERATOR_PLATFORM CMAKE_GENERATOR_TOOLSET
+    CMAKE_PLAT CMAKE_PREFIX_PATH CONDA_BUILD_CROSS_COMPILATION DISTUTILS_USE_SDK INCLUDE LATEST_VS
+    LIB MSSdk MSYS2_ARG_CONV_EXCL MSYS2_ENV_CONV_EXCL NEWER_VS_WITH_OLDER_VC PY_VCRUNTIME_REDIST
+    USE_NEW_CMAKE_GEN_SYNTAX VCVARSBAT VSINSTALLDIR VS_MAJOR VS_VERSION VS_YEAR WindowsSDKVer
```

Comment:
Let's unset things like `USE_NEW_CMAKE_GEN_SYNTAX` in the activate script itself.

Reply:
Good idea. I made an attempt in https://github.com/conda-forge/vc-feedstock/pull/93/commits/39eec1093883c65079484e14c0538457527d022a based on what I think are purely local variables. Let me know if I got one wrong.